system/lib/wasmfs/backends/opfs_backend.cpp

```diff
@@ -0,0 +1,279 @@
+// Copyright 2022 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#include "backend.h"
+#include "file.h"
+#include "support.h"
+#include "thread_utils.h"
+#include "wasmfs.h"
+#include <stdlib.h>
+
+namespace wasmfs {
```

Comment:
Is the C++ stuff in this file designed to be used outside of this TU?  Perhaps use anonymous namespace instead?

Reply:
I switched this to an anonymous namespace and added a `using namespace wasmfs;` to the top of the file to keep name resolution working.